File: kitty-transcoder/kitty-transcoder-api/src/main/java/icu/jiapeng/kitty/transcoder/api/TaskVO.java
package icu.jiapeng.kitty.transcoder.api;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
public class TaskVO {

    @Schema(description = "任务ID", example = "task_123456")
    private String id;

    @Schema(description = "任务状态", example = "PROCESSING")
    private String status;

    @Schema(description = "转码进度", example = "50")
    private Integer progress;

    @Schema(description = "输入类型 DISK/HTTP")
    private String inputType;

    @Schema(description = "输入路径或URL", example = "/path/to/video.mp4")
    private String inputPath;

    @Schema(description = "输入文件路径（兼容）", example = "/path/to/video.mp4")
    private String inputFile;

    @Schema(description = "输出磁盘路径", example = "/path/to/output.mp4")
    private String outputPath;

    @Schema(description = "输出文件路径（兼容）", example = "/path/to/output.mp4")
    private String outputFile;

    @Schema(description = "输出 HTTP 地址")
    private String outputHttpUrl;

    @Schema(description = "策略ID", example = "strategy_001")
    private String strategyId;

    @Schema(description = "水印地址（创建时传入）")
    private String watermarkUrl;
    @Schema(description = "水印位置")
    private String watermarkPosition;

    @Schema(description = "创建时间", example = "1739457045000")
    private Long createdAt;

    @Schema(description = "开始时间", example = "1739457050000")
    private Long startedAt;

    @Schema(description = "完成时间", example = "1739457345000")
    private Long completedAt;

    @Schema(description = "错误信息", example = "转码失败：FFmpeg 执行错误")
    private String errorMessage;
}

File: kitty-transcoder-go/internal/model/task_vo.go
package model

// TaskVO 与 Java TaskVO 对齐，API 返回用
type TaskVO struct {
	ID                string               `json:"id"`
	TaskType          string               `json:"taskType"`
	Status            string               `json:"status"`
	Progress          int                  `json:"progress"`
	InputType         string               `json:"inputType"`
	InputPath         string               `json:"inputPath"`
	InputFile         string               `json:"inputFile"`
	OutputPath        string               `json:"outputPath"`
	OutputFile        string               `json:"outputFile"`
	OutputHttpURL     string               `json:"outputHttpUrl"`
	StrategyID        int64                `json:"strategyId"`
	WatermarkURL      string               `json:"watermarkUrl"`
	WatermarkPosition string               `json:"watermarkPosition"`
	Notifications     []NotificationConfig `json:"notifications,omitempty"`
	CreatedAt         int64                `json:"createdAt"`
	StartedAt         int64                `json:"startedAt,omitempty"`
	CompletedAt       int64                `json:"completedAt,omitempty"`
	ErrorMessage      string               `json:"errorMessage,omitempty"`
}

// NotificationConfig 与 Java NotificationConfig 对齐（API 用 method/target，兼容 DB 中的 type）
type NotificationConfig struct {
	Method string `json:"method"`
	Target string `json:"target"`
	Type   string `json:"type,omitempty"` // 兼容旧数据：解析后填到 Method
}

// ListTasksResponse 分页任务列表响应，与 Java 对齐（含分页信息）
type ListTasksResponse struct {
	List     []TaskVO `json:"list"`
	Total    int64    `json:"total"`
	Page     int      `json:"page"`
	PageSize int      `json:"pageSize"`
}
